daemon/commit.go

```diff
@@ -74,6 +74,7 @@ func merge(userConf, imageConf *containertypes.Config) error {
 	if len(userConf.Entrypoint) == 0 {
 		if len(userConf.Cmd) == 0 {
 			userConf.Cmd = imageConf.Cmd
+			userConf.ArgsEscaped = imageConf.ArgsEscaped
```

Comment:
I'm a bit confused by the original purpose behind this flag. It seems that by the time we get to daemon.CreateSpec() if the args are not escaped, we escape them - and this is just for windows. Why aren't we just always escaping the args in the first place (on windows) whenever we stash them in the Cmd and EntryPoint arrays - then we don't need to have this flag at all and we can always just blindly copy the data into the spec.  I must be missing something.